src/app/pages/viewing/messages/messages.component.ts

```diff
@@ -13,11 +16,50 @@ import { Fonts } from '../../../shared/interfaces';
 export class MessagesComponent implements OnInit {
   fonts: Fonts;
   messageToDelete: string;
-
-  constructor(public viewService: ViewingService, public authService: AuthService) { }
+  constructor(public viewService: ViewingService, public authService: AuthService, private toastr: ToastrService) { }
 
   ngOnInit(): void {
     this.fonts = FontService.fonts;
+    this.viewService.nowMillis = Timestamp.now().toMillis();
+  }
+
+  /**
+   * Generate the lightness value of HSL from RBG
+   */
+  getLightness(r, g, b) {
+    r /= 255, g /= 255, b /= 255;
+    const max = Math.max(r, g, b), min = Math.min(r, g, b);
+    const l = (max + min) / 2;
+    return l;
+  }
+
+  /**
+   * Generate a darkened version of backColor, returning this new header color and the appropriate header text color
+   * @param backColor: background color of the message
+   */
+  generateHeaderColor(backColor: string) {
+    const bgR = parseInt(backColor.slice(1, 3), 16);
+    const bgG = parseInt(backColor.slice(3, 5), 16);
+    const bgB = parseInt(backColor.slice(5), 16);
+    const hR = bgR + 20 > 255 ? 255 : bgR + 20;
+    const hG = bgG + 20 > 255 ? 255 : bgG + 20;
+    const hB = bgB + 20 > 255 ? 255 : bgB + 20;
+    const headerColor = hR.toString(16) + hG.toString(16) + hB.toString(16);
+    let headerTextColor;
+    if (this.getLightness(hR, hG, hB) > 0.65) {
```

Comment:
You can play with the value, but right now it's too large and is rendering text unreadable:
![image](https://user-images.githubusercontent.com/39015447/86126438-561b6a80-baac-11ea-950b-038206e6c1ee.png)
![image](https://user-images.githubusercontent.com/39015447/86126452-5c114b80-baac-11ea-877e-f9b3594bf695.png)
```suggestion
    if (this.getLightness(hR, hG, hB) > 0.5) {
```